### PR TITLE
MAINT: Update required cython version to 0.29.14.

### PR DESCRIPTION
**Pull request by charris**
This is needed for the latest fixes for Python 3.8.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 requires = [
     "setuptools",
     "wheel",
-    "Cython>=0.29.13",  # Note: keep in sync with tools/cythonize.py
+    "Cython>=0.29.14",  # Note: keep in sync with tools/cythonize.py
```
Comment (rgommers) [inline pyproject.toml]: see the comment on this line: please update `cythonize.py` as well

**Comment by mattip** [comment]
interesting that the `test_requirements.txt` already specified 0.29.14.

File: pyproject.toml (hunk 1 above)
Comment (charris) [reply pyproject.toml]: Done.

**Comment by rgommers** [comment]
Merged, thanks Chuck

**Pull request merged**